### PR TITLE
Updated Regional PD failover test to use node taints instead of instance group deletion

### LINKED ISSUE #69005: RePD failover test flaking
**Is this a BUG REPORT or FEATURE REQUEST?**:
@kubernetes/sig-storage-bugs 

**What happened**:
Testgrid: https://k8s-testgrid.appspot.com/sig-storage#gke-regional-serial

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/regional_pd.go:81
Error getting instance group gke-bootstrap-e2e-default-pool-b9ea428e-grp in zone us-central1-c
Expected error:
    <*googleapi.Error | 0xc422a16b90>: {
        Code: 404,
        Message: "The resource 'projects/gke-up-g1-3-clat-up-mas/zones/us-central1-c/instanceGroups/gke-bootstrap-e2e-default-pool-b9ea428e-grp' was not found",
        Body: "{\n \"error\": {\n  \"errors\": [\n   {\n    \"domain\": \"global\",\n    \"reason\": \"notFound\",\n    \"message\": \"The resource 'projects/gke-up-g1-3-clat-up-mas/zones/us-central1-c/instanceGroups/gke-bootstrap-e2e-default-pool-b9ea428e-grp' was not found\"\n   }\n  ],\n  \"code\": 404,\n  \"message\": \"The resource 'projects/gke-up-g1-3-clat-up-mas/zones/us-central1-c/instanceGroups/gke-bootstrap-e2e-default-pool-b9ea428e-grp' was not found\"\n }\n}\n",
        Header: nil,
        Errors: [
            {
                Reason: "notFound",
                Message: "The resource 'projects/gke-up-g1-3-clat-up-mas/zones/us-central1-c/instanceGroups/gke-bootstrap-e2e-default-pool-b9ea428e-grp' was not found",
            },
        ],
    }
    googleapi: Error 404: The resource 'projects/gke-up-g1-3-clat-up-mas/zones/us-central1-c/instanceGroups/gke-bootstrap-e2e-default-pool-b9ea428e-grp' was not found, notFound
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/regional_pd.go:219
```

### PR DESCRIPTION
**Pull request by verult**
**What this PR does / why we need it**: The existing failover mechanism (deleting GCE managed instance groups) relies on the wrong NodeInstanceGroup parameter format from the test context for regional clusters, and there is no simple way to continue using the instance group technique.

This PR fixes the problem and also addresses a previous TODO left in the test by converting to tainting nodes instead.

Need to be cherry-picked to 1.12

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #69005 

**Release note**:

```release-note
NONE
```

/sig storage
/assign @msau42

### CODE REVIEW

File: test/e2e/storage/regional_pd.go [hunk 1]
```diff
@@ -201,41 +203,42 @@ func testZonalFailover(c clientset.Interface, ns string) {
...
 
 	defer func() {
-		framework.Logf("recreating instance group %s", instanceGroup.Name)
-
-		framework.ExpectNoError(framework.CreateManagedInstanceGroup(instanceGroup.Size, podZone, templateName),
-			"Error recreating instance group %s in zone %s", instanceGroup.Name, podZone)
-		framework.ExpectNoError(framework.WaitForReadyNodes(c, nodeCount, framework.RestartNodeReadyAgainTimeout),
-			"Error waiting for nodes from the new instance group to become ready.")
+		framework.Logf("removing previously added node taints")
+		nodesInZone, err = c.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: selector.String()})
```
Comment (msau42) [inline test/e2e/storage/regional_pd.go]: do you need to call this again or can you reuse the nodes list from above?  Any error here could skip the taint removal.

File: test/e2e/storage/regional_pd.go [hunk 2]
```diff
@@ -259,21 +261,42 @@ func testZonalFailover(c clientset.Interface, ns string) {
...
+func removeTaint(c clientset.Interface, nodes []v1.Node) {
+	for _, node := range nodes {
+		var newTaints []v1.Taint
+		for _, taint := range node.Spec.Taints {
+			if taint.Key != taintKey {
+				newTaints = append(newTaints, taint)
+			}
+		}
+		node.Spec.Taints = newTaints
+		_, err := c.CoreV1().Nodes().Update(&node)
```
Comment (msau42) [inline test/e2e/storage/regional_pd.go]: you might need to retry the update a few times.  Node currently does a heartbeat every few seconds by updating the node object.

File: test/e2e/storage/regional_pd.go [hunk 3]
```diff
@@ -44,6 +48,7 @@ import (
 const (
 	pvDeletionTimeout       = 3 * time.Minute
 	statefulSetReadyTimeout = 3 * time.Minute
+	taintKey                = "zoneTaint"
```
Comment (msau42) [inline test/e2e/storage/regional_pd.go]: I would make this key name more specific to make it easier to debug. Maybe use the test namespace name?

File: test/e2e/storage/regional_pd.go (hunk 1 above)
Comment (verult) [reply test/e2e/storage/regional_pd.go]: Yep the list can be reused. Thanks for the catch!

**Comment by verult** [comment]
Addressed comments

**Pull request merged**